addons/test_mail/tests/test_mail_mail.py

```diff
@@ -823,6 +830,97 @@ def test_mail_mail_values_unicode(self):
+                    else 'attachment' in message.get('Content-Disposition', ''))
+
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
+        # Define some constant to ease the understanding of the test
+        test_mail_server = self.mail_server_domain_2
+        cst_max_size_always_exceed = 0.1
+        cst_max_size_never_exceed = 10
+
+        # cst_max... are used to define max_email_size on the server that is not used with value that would
+        # lead to an incorrect expected_is_links if they were used. We do this to check that the right max
+        # email size is used.
+        for (n_attachment, mail_server, default_email_size, max_email_size, business_attachment), expected_is_links in (
+                # 1 attachment which doesn't exceed max size
+                ((1, self.env['ir.mail_server'], 0.3, cst_max_size_always_exceed, True), False),
+                # 3 attachment: exceed max size
+                ((3, self.env['ir.mail_server'], 0.3, cst_max_size_never_exceed, True), True),
+                # 1 attachment: exceed max size
+                ((1, self.env['ir.mail_server'], 0.1, cst_max_size_never_exceed, True), True),
+                # Same as above with a specific server. Note that the default and server max_email size are reversed.
+                ((1, test_mail_server, cst_max_size_always_exceed, 0.3, True), False),
+                ((3, test_mail_server, cst_max_size_never_exceed, 0.3, True), True),
+                ((1, test_mail_server, cst_max_size_never_exceed, 0.1, True), True),
+                # Attachments not linked to a business record are never turned to link
+                ((3, self.env['ir.mail_server'], 0.3, cst_max_size_never_exceed, False), False),
+                ((1, test_mail_server, cst_max_size_never_exceed, 0.1, False), False),
+        ):
+            self.env['ir.config_parameter'].sudo().set_param('base.default_max_email_size', default_email_size)
+            if mail_server:
+                mail_server.max_email_size = max_email_size
+            attachments = self.env['ir.attachment'].sudo().create([{
+                'name': f'attachment{idx_attachment}',
+                'res_name': 'test',
+                'res_model': self.test_record._name if business_attachment else 'mail.message',
+                'res_id': self.test_record.id if business_attachment else 0,
+                'datas': datas,
+            } for idx_attachment in range(n_attachment)])
+            with self.mock_smtplib_connection():
+                mails = self.env['mail.mail'].create([{
+                    'attachment_ids': attachments.ids,
+                    'body_html': '<p>Test</p>',
+                    'email_from': 'test@test_2.com',
+                    'email_to': f'u_{mail_idx}_s-{bool(mail_server)}_a-{n_attachment}_l-{expected_is_links}@test.com',
```

Comment:
Would rather add more parameters to the subTest than having this obscure string that summarizes everything ^^